ui/components/src/Legend/ListLegend.tsx

```diff
@@ -30,140 +29,40 @@ export interface ListLegendProps {
   initialRowHeight?: number;
 }
 
-const DEFAULT_INITIAL_ROW_HEIGHT = 26;
-
 /**
  * ListLegend is used when legend.position is 'right' since legend items are
  * stacked. It is also used for `bottom` positioned legends when there are a
  * large number of items because it is virtualized and easier to visually scan
  * large numbers of items when there is a single item per row.
  */
-export function ListLegend({ items, height, width, initialRowHeight = DEFAULT_INITIAL_ROW_HEIGHT }: ListLegendProps) {
-  // Storing a ref to the react-window `VariableSizeList`, so we can call
-  // `resetAfterIndex` to resize the list after mouseover/out events to account
-  // for the change in list items on hover.
-  const listRef = useRef<VariableSizeList>(null);
-  // Storing row heights, so we can use dynamic heights, which enables the
-  // user the hover to show the full label, while still having a virtualized
-  // list.
-  const rowHeights = useRef<Record<number, number>>({});
-
+export function ListLegend({ items, height, width }: ListLegendProps) {
   const theme = useTheme();
-  // Padding value used throughout to adjust the react-window virtual layouts
-  // to simulate padding per the guidance from:
-  // https://github.com/bvaughn/react-window#can-i-add-padding-to-the-top-and-bottom-of-a-list
+  // Padding value used in the react virtuoso header/footer components to
+  // simulate top/bottom padding based on recommendation in this
+  // issue.
+  // https://github.com/petyosi/react-virtuoso/issues/238
   const LIST_PADDING = parseInt(theme.spacing(1), 10);
+  const mockPadding = <Box sx={{ width: '100%', height: `${LIST_PADDING}px` }}></Box>;
 
   // show full labels on hover when there are many total series
   const truncateLabels = items.length > 5;
 
-  // Gets the row height for a given item to enable the virtualized list to
```

Comment:
This code is waaaay simpler because react-virtuoso has dynamic height handling baked inside the library, so you don't have to do it yourself. I probably should have done a teeny bit more looking ahead to this table work and considered this lib instead when I did the virtualization for the list, but hindsight is always 20/20. 🤷‍♀️